### PR TITLE
Master setting list and detail api.

### PR DESCRIPTION
**Pull request by damini-7span**

### CODE REVIEW

File: app/Http/Controllers/Api/V1/MasterSettingController.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    type: 'string'
+                )
+            ),
+        ],
+    )]
+    public function show($id)
+    {
+        $masterSettingObj = $this->masterSettingService->resource($id);
+
+        return isset($masterSettingObj['errors']) ? $this->error($masterSettingObj) : new Resource($masterSettingObj);
```
Comment (harshil-7span) [inline app/Http/Controllers/Api/V1/MasterSettingController.php:92]: @damini-7span If we have throw the error than we call the exception from the service file, So no need to check the error here.

File: database/migrations/2024_07_09_122245_create_master_settings_table.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('master_settings', function (Blueprint $table) {
+            $table->id();
+            $table->string('key', 255);
```
Comment (harshil-7span) [inline database/migrations/2024_07_09_122245_create_master_settings_table.php:13]: @damini-7span Please add the index on key fields

File: routes/api-v1.php [hunk 3]
```diff
@@ -24,3 +25,4 @@
 
 Route::post('generate-signed-url', SignedUrlController::class);
 Route::get('countries', CountryController::class);
+Route::apiResource('versions', MasterSettingController::class)->only(['index', 'show']);
```
Comment (harshil-7span) [inline routes/api-v1.php:28]: @damini-7span Please add this route in admin.php route file, And it's access after the login.

Comment (damini-7span) [reply routes/api-v1.php:28]: As we discussed we decided to take is_public field in database table and remain route in api-v1.php as it is.

**Review by harshil-7span — APPROVED** [review]

**Pull request merged**